packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts

```diff
@@ -1249,4 +1256,32 @@ export class AmplifyAuthCognitoStack extends cdk.Stack implements AmplifyAuthCog
       });
     });
   };
+
+  createHostedUIDomainResource = (props: CognitoStackOptions) => {
+    if (!props.hostedUIDomainName) {
+      return;
+    }
+
+    const authCfnTemplatePath = getResourceCfnTemplatePath(getBackendDirPath(), 'auth', props.resourceName);
+    const authCfnTemplate: Template | undefined = readJson(authCfnTemplatePath, { throwIfNotExist: false });
+    const lambdaCalloutCreatedInCloud = authCfnTemplate?.Resources?.HostedUICustomResource?.Type === 'AWS::Lambda::Function';
+    const userPoolDomainCreatedInCloud = authCfnTemplate?.Resources?.HostedUIDomainResource?.Type === 'AWS::Cognito::UserPoolDomain';
+
+    if (lambdaCalloutCreatedInCloud && !userPoolDomainCreatedInCloud) {
+      this.deleteExistingHostedUICustomResource();
```

Comment:
Can a case happen , when project has lambdaCalloutCreatedInCloud nut the cfn template isn't created yet ?

Reply:
I don't think so, but we should check